src/v/utils/moving_average.h

```diff
@@ -50,6 +51,35 @@ class moving_average {
     size_t _valid_samples{0};
 };
 
+template<typename T, size_t Samples>
+class moving_max {
+public:
+    explicit moving_max(T initial_value)
+      : _running_max(initial_value) {
+        std::fill(_samples.begin(), _samples.end(), initial_value);
+    }
+
+    void update(T v) {
+        // Value at '_ids' will be phased out. If it's equal to current
+        // max we need to recalculate it.
+        bool need_recalc = _running_max == _samples[_idx];
+        _samples[_idx] = v;
+        _idx = (_idx + 1) % Samples;
+        if (v > _running_max) {
```

Comment:
nit: this could be `v >= running_max` to avoid a full iteration when `v == _running_max`

Reply:
the code is removed because I decided not to use history